nifi-nar-bundles/nifi-extension-utils/nifi-database-utils/src/main/java/org/apache/nifi/util/db/JdbcCommon.java

```diff
@@ -711,18 +711,59 @@ public static void setParameters(final PreparedStatement stmt, final Map<String,
         }
     }
 
+    /**
+     * Sets all of the appropriate parameters on the given PreparedStatement, based on the given FlowFile attributes.
+     *
+     * @param stmt the statement to set the parameters on
+     * @param attributes the attributes from which to derive parameter indices, values, and types
+     * @throws SQLException if the PreparedStatement throws a SQLException when the appropriate setter is called
+     */
+    public static void setSensitiveParameters(final PreparedStatement stmt, final Map<String, SensitiveValueWrapper> attributes) throws SQLException {
```

Comment:
This and the old `setParameters()` methods have mostly the same logic, except the masked logging.
To avoid code duplication, would it be possible the extract the common parts?

The simplest (but maybe not the best) approach would be to convert `Map<String, String>` to `Map<String, SensitiveValueWrapper>` (with fixed `sensitive=false`) in the old method and then to call `setSensitiveParameters()` with that map.

Reply:
If the old method is not used, I think it makes more sense to delete the old method.

Reply:
- According to @turcsanyip's remarks, since logging occurs only in the setParameter method if the flowfile attribute is "sql.arguments.*type," and
- @exceptionfactory - filtering out the type is not needed
we may revert to the old behavior.